subprojects/logging/src/main/java/org/gradle/internal/logging/console/WorkInProgressRenderer.java

```diff
@@ -208,4 +176,48 @@ void renderNow() {
             label.setText(labelFormatter.format(operation));
         }
     }
+
+    private static class ParentRegistry {
+        private final Map<OperationIdentifier, Set<OperationIdentifier>> parentIdToChildrenIds = new HashMap<OperationIdentifier, Set<OperationIdentifier>>();
+
+        public ChildrenRegistry find(final OperationIdentifier parentId) {
+            return new ChildrenRegistry() {
+                @Override
+                public void add(OperationIdentifier childId) {
+                    Set<OperationIdentifier> children = parentIdToChildrenIds.get(parentId);
+                    if (children == null) {
+                        children = new HashSet<OperationIdentifier>();
+                        parentIdToChildrenIds.put(parentId, children);
+                    }
+                    children.add(childId);
+                }
+
+                @Override
+                public void remove(OperationIdentifier childId) {
+                    Set<OperationIdentifier> children = parentIdToChildrenIds.get(parentId);
+                    if (children == null) {
+                        throw new IllegalStateException("");
```

Comment:
:x: We should figure out how this'll work given my changes. We may just want to `return;` here.